pkg/agent/endpoints/endpoints_test.go

```diff
@@ -332,3 +330,13 @@ func logEntryWithPID(level logrus.Level, msg string, keyvalues ...interface{}) s
 	}
 	return spiretest.LogEntry{Level: level, Message: msg, Data: data}
 }
+
+func waiForListening(t *testing.T, e *Endpoints) {
+	timer := time.NewTimer(time.Second)
```

Comment:
I wonder if we should just rely on our global test timeout (1 minute). In either case, one second may be too small for bogged down CI/CD runners. Also, I think we should take the error channel in here and select on that (maybe instead of the timer). We expect either the hook to fire or ListenAndServe to fail.

Reply:
That's a great idea!